src/deadline/client/api/_job_monitoring.py

```diff
@@ -0,0 +1,322 @@
+        start_time: Optional start time for logs as a datetime object.
+        end_time: Optional end time for logs as a datetime object.
+        next_token: Optional token for pagination of results.
+        config: Optional configuration object.
+
+    Returns:
+        A SessionLogResult object containing the log events and metadata.
+
+    Raises:
+        DeadlineOperationError: If there's an error retrieving the logs.
+    """
+    # Get the Deadline client to use for getting queue credentials
+    deadline = get_boto3_client("deadline", config=config)
+
+    # Check if we have user and identity store ID (from Deadline Cloud monitor)
+    user_id, identity_store_id = get_user_and_identity_store_id(config=config)
+
+    # Create logs client - either with queue credentials or directly
+    if user_id and identity_store_id:
+        # Get a session with queue user credentials
+        try:
+            queue_session = get_queue_user_boto3_session(
+                deadline=deadline, config=config, farm_id=farm_id, queue_id=queue_id
+            )
+            logs_client = queue_session.client("logs")
+        except Exception as e:
+            raise DeadlineOperationError(f"Failed to get queue credentials: {e}")
+    else:
+        # Use the same boto session as for deadline
+        logs_client = get_boto3_client("logs", config=config)
+
+    # Construct the log group name
+    log_group_name = f"/aws/deadline/{farm_id}/{queue_id}"
+
+    # Ensure session ID has the correct format
+    if not session_id.startswith("session-"):
+        session_id = f"session-{session_id}"
+
+    # Prepare parameters for GetLogEvents
+    params = {
+        "logGroupName": log_group_name,
+        "logStreamName": session_id,
+        "limit": limit,
+        "startFromHead": False,  # Get the most recent logs first
+    }
+
+    # Add next_token if provided
+    if next_token:
+        params["nextToken"] = next_token
+
+    # Add optional time parameters if provided
+    if start_time:
+        try:
+            # Convert datetime to milliseconds since epoch
+            start_timestamp = int(start_time.timestamp() * 1000)
+            params["startTime"] = start_timestamp
+        except (ValueError, AttributeError) as e:
+            raise DeadlineOperationError(f"Invalid start time: {e}")
+
+    if end_time:
+        try:
+            # Convert datetime to milliseconds since epoch
+            end_timestamp = int(end_time.timestamp() * 1000)
+            params["endTime"] = end_timestamp
+        except (ValueError, AttributeError) as e:
+            raise DeadlineOperationError(f"Invalid end time: {e}")
+
+    try:
+        response = logs_client.get_log_events(**params)
+
+        # Convert to strongly typed objects
+        events = []
+        for event in response.get("events", []):
+            events.append(
+                LogEvent(
+                    timestamp=datetime.datetime.fromtimestamp(event["timestamp"] / 1000),
```

Comment:
From https://docs.python.org/3/library/datetime.html#datetime.datetime.fromtimestamp, it looks like this creates the timestamp in the local timezone, not UTC. should this be:

```
datetime.datetime.fromtimestamp(event["timestamp"] / 1000, tz=datetime.tz.utc)
```